manager/src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,410 @@
+package io.gingersnapproject.k8s;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.fabric8.kubernetes.client.informers.SharedIndexInformer;
+import io.gingersnapproject.configuration.Connector;
+import io.gingersnapproject.configuration.EagerRule;
+import io.gingersnapproject.configuration.LazyRule;
+import io.gingersnapproject.k8s.configuration.KubernetesConfiguration;
+import io.gingersnapproject.proto.api.config.v1alpha1.EagerCacheRuleSpec;
+import io.gingersnapproject.proto.api.config.v1alpha1.LazyCacheRuleSpec;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+/** Creates two informer on k8s configMap for Lazy and Eager Gingersnap rules
+ * ConfigMap names come from properties
+ * gingersnap.k8s.eager-config-map
+ * gingersnap.k8s.lazy-config-map
+*/
+@ApplicationScoped
+public class CacheRuleInformer {
+   @Inject
+   Instance<KubernetesClient> client;
+   @Inject
+   KubernetesConfiguration configuration;
+   @Inject
+   Event<RuleEvents.LazyRuleAdded> lazyEventAdd;
+   @Inject
+   Event<RuleEvents.LazyRuleRemoved> lazyEventRemove;
+   @Inject
+   Event<RuleEvents.EagerRuleAdded> eagerEventAdd;
+   @Inject
+   Event<RuleEvents.EagerRuleRemoved> eagerEventRemove;
```

Comment:
Splitting the Events into a dedicated class, similar to the db-syncer `NotificationManager`, would simplify the code a lot here as you would only need to pass one instance to the various dispatchers and you wouldn't need to have all of the `Event` variables defined. It could just be package-private for this use-case, but IMO it would improve readability.

Reply:
I've checked: I can split the class but we still need an instance for each specific event, also NotificationManager does the [same](https://github.com/gingersnap-project/db-syncer/blob/f5405abd3de433d05455af93945143d4e053280b/src/main/java/io/gingersnapproject/cdc/event/NotificationManager.java#L24-L33)

Reply:
Yeah we will still need to inject the event in the `NoficationManager` equivalent, but it means we have a lot less variables to pass to the various inner classes in `CacheRuleInformer` as we can just pass the manager.

Reply:
oh didn't noticed that NotificationManager was an util class. ok